core/base/assignmentSolver/AssignmentAuction.h

```diff
@@ -0,0 +1,356 @@
+/// \ingroup base
+/// \class ttk::AssignmentAuction
+/// \author Mathieu Pont (mathieu.pont@lip6.fr)
+///
+/// Auction algorithm for Balanced and Unbalanced Assignement Problem
+///
+/// For the unbalanced problem:
+///   The cost matrix in input has a size of (n + 1) x (m + 1)
+///   - n is the number of jobs, m the number of workers
+///   - the nth row contains the cost of not assigning workers
+///   - the mth column is the same but with jobs
+///   - the last cell (costMatrix[n][m]) is not used
+
+#ifndef _ASSIGNMENTAUCTION_H
+#define _ASSIGNMENTAUCTION_H
+
+#include <Debug.h>
+
+#include "AssignmentSolver.h"
+
+namespace ttk {
+
+  template <class dataType>
+  class AssignmentAuction : virtual public Debug,
+                            public AssignmentSolver<dataType> {
+
+  public:
+    AssignmentAuction(){};
+
+    ~AssignmentAuction(){};
+
+    int run(std::vector<asgnMatchingTuple> &matchings);
+    void runAuctionRound(std::vector<std::vector<dataType>> &cMatrix);
+
+    void initFirstRound();
+    void initBiddersAndGoods();
+    void initEpsilon();
+    void epsilonScaling();
+    void makeBalancedMatrix(std::vector<std::vector<dataType>> &matrix);
+
+    bool stoppingCriterion(std::vector<std::vector<dataType>> &cMatrix);
+    dataType getRelativePrecision(std::vector<std::vector<dataType>> &cMatrix);
+    dataType getMatchingDistance(std::vector<std::vector<dataType>> &cMatrix);
+
+    inline void setBalanced(bool balanced) {
+      AssignmentSolver<dataType>::setBalanced(balanced);
+      if(this->balancedAssignment)
+        goodPrices.resize(this->colSize, 0);
+      else
+        goodPrices.resize((this->colSize - 1) + (this->rowSize - 1), 0);
+    }
+
+    inline void setNumberOfRounds(int noRounds) {
+      numberOfRounds = noRounds;
+    }
+
+    inline int getIter() {
+      return iter;
+    }
+
+    inline void setEpsilon(double eps) {
+      epsilon = eps;
+    }
+
+    inline void setEpsilonDiviserMultiplier(double div) {
+      epsilonDiviserMultiplier = div;
+    }
+
+    inline void setPrices(std::vector<double> prices) {
+      goodPrices.resize(prices.size(), 0);
+      for(unsigned int i = 0; i < prices.size(); ++i)
+        goodPrices[i] = prices[i];
+    }
+
+    inline std::vector<double> &getPrices() {
+      return goodPrices;
+    }
+
+    template <class vecType>
+    void printVector(std::vector<vecType> &vec) {
+      std::stringstream ss;
+      for(auto valTemp : vec)
+        ss << valTemp << " ";
+      printMsg(ss.str(), debug::Priority::VERBOSE);
+      printMsg(" ------------------------------ ", debug::Priority::VERBOSE);
```

Comment:
actually, you have pre-specified separators available for you. instead of this line, you can use for instance:
``this->printMsg(ttk::debug::Separator::L1);`` which will have a similar effect (but with a better integration in the overall logs). checkout the ``Debug.h`` file to see the available separators. if you could spread that through the PR, that'd be awesome.

Reply:
Ok thanks for these information about printMsg, I will fix this in the next commit!